eclog/src/test/java/com/ericsson/bss/cassandra/ecaudit/eclog/TestOptionParser.java

```diff
@@ -0,0 +1,351 @@
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.ericsson.bss.cassandra.ecaudit.eclog;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.nio.file.Paths;
+
+import org.apache.commons.cli.ParseException;
+import org.junit.Test;
+
+import net.openhft.chronicle.queue.RollCycles;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+
+public class TestOptionParser
+{
+    private OptionParser parser = new OptionParser();
+
+    @Test
+    public void noOptions()
+    {
+        String[] argv = givenInputOptions();
+
+        assertThatExceptionOfType(ParseException.class)
+        .isThrownBy(() -> parser.parse(argv))
+        .withMessageContaining("directory");
+    }
+
+    @Test
+    public void withDirectory() throws ParseException
+    {
+        String[] argv = givenInputOptions("./dir");
+
+        ToolOptions options = parser.parse(argv);
+
+        assertThat(options.path()).isEqualTo(Paths.get("./dir"));
+        assertThat(options.follow()).isFalse();
+        assertThat(options.rollCycle().isPresent()).isFalse();
+        assertThat(options.limit().isPresent()).isFalse();
+        assertThat(options.tail().isPresent()).isFalse();
+    }
+
+    @Test
+    public void withFollowAndDirectory() throws ParseException
+    {
+        String[] argv = givenInputOptions("-f", "./dir");
+
+        ToolOptions options = parser.parse(argv);
+
+        assertThat(options.path()).isEqualTo(Paths.get("./dir"));
+        assertThat(options.follow()).isTrue();
+        assertThat(options.rollCycle().isPresent()).isFalse();
+        assertThat(options.limit().isPresent()).isFalse();
+        assertThat(options.tail().isPresent()).isFalse();
+    }
+
+    @Test
+    public void withLongFollowAndDirectory() throws ParseException
+    {
+        String[] argv = givenInputOptions("--follow", "./dir");
+
+        ToolOptions options = parser.parse(argv);
+
+        assertThat(options.path()).isEqualTo(Paths.get("./dir"));
+        assertThat(options.follow()).isTrue();
+        assertThat(options.rollCycle().isPresent()).isFalse();
+        assertThat(options.limit().isPresent()).isFalse();
+        assertThat(options.tail().isPresent()).isFalse();
+    }
+
+    @Test
+    public void withFollowAndNoDirectory()
+    {
+        String[] argv = givenInputOptions("-f");
+
+        assertThatExceptionOfType(ParseException.class)
+        .isThrownBy(() -> parser.parse(argv))
+        .withMessageContaining("directory");
+    }
+
+    @Test
+    public void withRollMinutelyAndDirectory() throws ParseException
```

Comment:
A lot of test cases and many of them are very similar (contains a lot of similar asserts)... Are all needed to get full coverage?
Can there be like 3 testcase testing "all" parameters? 
- One with all params with given with name
- One with all params with given with long name
- One with no params given

then special test case for dir given and not...

Also maybe the private methods could be made protected final and then tested "directly" for the exception handling...